xrally_kubernetes/service.py

```diff
@@ -78,3 +154,75 @@ def list_namespaces(self):
                  "uid": r.metadata.uid,
                  "labels": r.metadata.labels}
                 for r in self.v1_client.list_namespace().items]
+
+    @atomic.action_timer("kube.create_namespace")
+    def create_namespace(self, name):
+        """Create namespace.
+
+        :param name: namespace name
+        """
+        manifest = {
+            "apiVersion": "v1",
+            "kind": "Namespace",
+            "metadata": {
+                "name": name,
+                "labels": {
+                    "role": "test-%s" % name
+                }
+            }
+        }
+        self.v1_client.create_namespace(body=manifest)
+        LOG.info("Namespace %s create started" % name)
+
+    @atomic.action_timer("create_namespace_and_wait_active_status")
+    def create_namespace_and_wait_active(self, name, sleep_time=5,
+                                         retries_total=30):
+        """Create namespace and wait until status phase won't be Active.
+
+        :param name: namespace name
+        :param sleep_time: poll interval between each two retries
+        :param retries_total: number of total retries of reading status
+        """
+        self.create_namespace(name)
+
+        LOG.debug("Wait until namespace %s status won't be active" % name)
+        wait_for_status(name,
+                        status="Active",
+                        read_method=self.v1_client.read_namespace,
+                        sleep_time=sleep_time,
+                        retries_total=retries_total)
+        LOG.info("Namespace %s has Active status now" % name)
+
+    @atomic.action_timer("kube.delete_namespace")
+    def delete_namespace(self, name):
+        """Delete namespace.
+
+        :param name: namespace name
+        """
+        resp = self.v1_client.delete_namespace(
+            name=name,
+            body=k8s_config.V1DeleteOptions()
+        )
+
+        LOG.info("Namespace %(name)s deleted. Status: %(status)s" % {
+            "name": name,
+            "status": resp.status
+        })
+
+    @atomic.action_timer("kube.delete_namespace_and_wait_termination")
+    def delete_namespace_and_wait_termination(self, name,
```

Comment:
same here about separate method and wrapping wait_for with a proper action timer.